furusiya/model/factories/monster_factory.py

```diff
@@ -4,11 +4,10 @@
 from model.helper_functions.death_functions import monster_death
 from model.components.fighter import Fighter
 from model.components.xp import XPComponent
-from model.entities.game_object import GameObject
 
 
-def create_monster(data, x, y, colour, name):
-    monster = GameObject(x, y, name[0], name, colour, blocks=True)
+def create_monster(data, x, y, colour, name, cls):
```

Comment:
I disagree with this. It forces us to create classes for every single monster, which I don't want. Ideally, monsters should be just composites of components; practically speaking, if we ever need to add a `SalamanderComponent` or `TreeSlimeComponent`, that's a smell, and we should come back and create classes for just those cases.

As a middle-ground -- although it's not needed here (and adding uneeded code adds complexity in terms of testing/maintenance), we can make the `cls` parameter optional and default to `GameObject`.

Reply:
The class is explicitly specified for every enemy in `generate_monsters`, which may very well be `GameObject`. I say this is good as-is.

Reply:
This is not okay. Every time I create a new class, I need to specify what should have been a very obvious choice for a base-class.